src/services/role/services/rolesService.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable no-param-reassign */
+const { NotFound } = require('@feathersjs/errors');
+const { RoleModel } = require('../model');
+
+// Do not proteced this route with authentication.
+const RoleServiceHooks = {
+	before: {
+		all: [],
+		find: [],
+		get: [],
+	},
+};
+
+const removeKeys = (keys = []) => (role) => {
+	keys.forEach((key) => {
+		delete role[key];
+	});
+	return role;
+};
+
+const paginate = (result, { $limit, $skip = 0 } = {}) => ({
+	total: result.length,
+	limit: $limit || result.length,
+	skip: $skip,
+	data: result.slice($skip, $limit),
+});
+
+const filterByQueryKey = (roles, key, value) => {
+	if (key.charAt(0) === '$') {
+		return roles;
+	}
+	return roles.filter((r) => r[key] === value);
+};
+
+const filterByQuery = (roles, query = {}) => {
+	let array = roles;
+	Object.keys(query).forEach((key) => {
+		array = filterByQueryKey(array, key, query[key]);
+	});
+	return array;
+};
+
+const getRoles = (query = {}) => RoleModel.find(query).lean().exec();
+
+/**
+ * This is a static services.
+ */
+class RoleService {
+	constructor({ docs } = {}) {
+		this.docs = docs || {};
+		this.err = Object.freeze({
+			solved: 'Can not solved the role',
+			load: 'Can not load roles from DB, or can not solved pre mutations.',
+		});
+		this.roles = undefined;
+		this.rolesDisplayName = Object.freeze({
+			teacher: 'Lehrer',
+			student: 'Schüler',
+			administrator: 'Administrator',
+			superhero: 'Schul-Cloud Admin',
+			demo: 'Demo',
+			demoTeacher: 'Demo',
+			demoStudent: 'Demo',
+			helpdesk: 'Helpdesk',
+			betaTeacher: 'Beta',
+			expert: 'Experte',
+		});
+		this.filterKeys = Object.freeze([]); // ['createdAt', 'updatedAt', 'roles', '__v']
+	}
+
+	async setup(app) {
+		this.app = app;
+		await this.init();
+	}
+
+	async init() {
+		this.roles = new Promise(async (resolve) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Promise executor functions should not be async.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5229889)